app/controllers/users_controller.rb

```diff
@@ -0,0 +1,105 @@
+class UsersController < ApplicationController
+  before_action :require_login, only: [:upvote]
+
+  def index
+    @users = User.all
+  end
+
+  def show
+    @user = User.find_by(id: params[:id])
+    return head :not_found if @user.nil?
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      flash[:success] = "#{@user.title} was successfully added!"
+      redirect_to User_path(@user.id)
+      return
+    else
+      flash.now[:error] = "#{@user.title} was not successfully added!"
+      render :new, status: :bad_request
+      return
+    end
+  end
+
+  def login_form
+    @user = User.new
+  end
+
+  def login
+    user = User.find_by(name: params[:user][:name])
+
+    if user.nil?
+      user = User.new(name: params[:user][:name])
+      if ! user.save
+        flash[:error] = "Unable to login"
+      end
```

Comment:
I'd generally use an `unless` here to be clearer:

```suggestion
      unless user.save
        flash[:error] = "Unable to login"
      end
```